services/plugins/plugins/validation.go

```diff
@@ -0,0 +1,25 @@
+package plugins
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/leanspace/terraform-provider-leanspace/helper"
+)
+
+func isValidSemVerForPlugins(i interface{}, fieldName string) (warnings []string, errorsOnField []error) {
+
+	warnings, errors := helper.IsValidSemVer(i, fieldName)
```

Comment:
just a question here but if it's `nil` does it fail or not?
It should be allowed for the simulations service

Reply:
If you have a resource with an empty sdkVersion : 
```
resource "leanspace_plugins" "test" {
  file_path                            = var.path
  type                                 = "COMMANDS_COMMAND_TRANSFORMER_PLUGIN_TYPE"
  implementation_class_name            = "io.myplugin.SimpleCommandTransformer"
  name                                 = "Terraform Command Transformer Plugin"
  description                          = "This is a plugin created through terraform!"
  source_code_file_download_authorized = true
  sdk_version                          = ""
}
```
You get this response : 
```
│ Error: expected "sdk_version" to be a valid semantic version MAJOR.MINOR.PATCH, got ""
│
│   with leanspace_plugins.test,
│   on main.tf line 43, in resource "leanspace_plugins" "test":
╷
│ Error: expected "sdk_version" to have major version between 1 and 2, got ""
│
│   with leanspace_plugins.test,
│   on main.tf line 43, in resource "leanspace_plugins" "test":
│   43:   sdk_version                          = ""
│
```

If you set the SdkVersion to nil : 
```
resource "leanspace_plugins" "test" {
  file_path                            = var.path
  type                                 = "COMMANDS_COMMAND_TRANSFORMER_PLUGIN_TYPE"
  implementation_class_name            = "io.myplugin.SimpleCommandTransformer"
  name                                 = "Terraform Command Transformer Plugin"
  description                          = "This is a plugin created through terraform!"
  source_code_file_download_authorized = true
  sdk_version                          = nil
}
```
You get this response : 
```
╷
│ Error: Invalid reference
│
│   on main.tf line 43, in resource "leanspace_plugins" "test":
│   43:   sdk_version                          = nil
│
│ A reference to a resource type must be followed by at least one attribute access, specifying the resource name.
```

If the SdkVersion is set to null, then the plugin is created : 
```
resource "leanspace_plugins" "test" {
  file_path                            = var.path
  type                                 = "COMMANDS_COMMAND_TRANSFORMER_PLUGIN_TYPE"
  implementation_class_name            = "io.myplugin.SimpleCommandTransformer"
  name                                 = "Terraform Command Transformer Plugin"
  description                          = "This is a plugin created through terraform!"
  source_code_file_download_authorized = true
  sdk_version                          = null
}
```

If you do not include the sdkVersion in the resource is works OK as well :
```
resource "leanspace_plugins" "test" {
  file_path                            = var.path
  type                                 = "COMMANDS_COMMAND_TRANSFORMER_PLUGIN_TYPE"
  implementation_class_name            = "io.myplugin.SimpleCommandTransformer"
  name                                 = "Terraform Command Transformer Plugin"
  description                          = "This is a plugin created through terraform!"
  source_code_file_download_authorized = true
}
```

Reply:
For simulations, the SdkVersion will be really set with a nil value or will it not be included in the resource configuration ?

Reply:
I think it's fine then, the sdkVersion should not be passed if you use simulation and if you do (because of a future update maybe) it still make sense to verify the semVer so it's good :)